sharding/peers.go

```diff
@@ -33,6 +35,54 @@ type peer struct {
 	address string
 }
 
+func (p *Peers) SmallestMin() (string, error) {
+	if len(p.peers) == 0 {
+		return "1", nil
+	}
+
+	peerList := make([]int, len(p.peers), len(p.peers))
+	i := 0
+	for peer := range p.peers {
+		num, err := strconv.Atoi(peer.shardID)
+		if err != nil {
+			return "", fmt.Errorf("Can't convert shardID %q to int", peer.shardID)
+		}
+
+		peerList[i] = num
+		i++
+	}
+
+	sort.Ints(peerList)
+
+	if peerList[0] != 1 {
```

Comment:
I don't think you need this? If `peerList` starts with e.g. 2, you'll enter the loop and 2 != (0+1) and you'll return 0+1 as expected.